src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/base.py

```diff
@@ -209,7 +209,19 @@ def _validate_operational_conditions(
         discharge_pressure = np.where(
             np.logical_and(np.min(rate, axis=0) <= 0, discharge_pressure <= 0), 1, discharge_pressure
         )
-
+        if not np.all(discharge_pressure >= suction_pressure):
+            logger.warning(
+                f"Inlet pressure needs to be a less than or equal to outlet pressure. Given values for inlet"
+                f" pressure: {suction_pressure.tolist()}. Given values for outlet pressure:"
+                f" {discharge_pressure.tolist()}. The affected time steps will not be calculated,"
+                f" and rate is set to zero."
+            )
+            rate = np.where(discharge_pressure < suction_pressure, 0, rate)
+        suction_pressure = np.where(
+            np.logical_and(np.min(rate, axis=0) <= 0, suction_pressure > discharge_pressure),
+            discharge_pressure,
+            suction_pressure,
+        )
```

Comment:
fairly complicated to read this numpy juggling. can it be extracted in a separate function and given some tests?

Reply:
Since we are now only validating input when rate != 0 the numpy juggling can be simplified a bit, it seems.